### PR TITLE
21c Docker image support

### PR DESCRIPTION
**Pull request by marcelo-ochoa**
This PR is for building latest Oracle EE and SE2 Docker image for RDBMS version 21c, using latest download: https://www.oracle.com/database/technologies/oracle-database-software-downloads.html
It is basically a clone of 19c scripts tested on Ubuntu 20.04 and Docker version 20.10.8
Note: There is no oracle-database-preinstall-19c rpm file at Oracle Linux 7 slim images, but it seem to work fine preparing the environment and dependency for 21c version.

### CODE REVIEW

**Comment by abhisbyk** [comment]
Raised the PR #2058  for supporting 21c Single Instance Database container image creation. It contains some additional things as compared to 19c. Please review.
@gvenzl

File: OracleDatabase/SingleInstance/dockerfiles/21.3.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#
+# HOW TO BUILD THIS IMAGE
+# -----------------------
+# Put all downloaded files in the same directory as this Dockerfile
+# Run: 
+#      $ docker build -t oracle/database:19.3.0-${EDITION} . 
+#
+# Pull base image
+# ---------------
+FROM oraclelinux:8-slim as base
```
Comment (Djelibeybi) [inline OracleDatabase/SingleInstance/dockerfiles/21.3.0/Dockerfile:23]: Please do not use `oraclelinux:8-slim`, as per the base image rules in the contribution guide: https://github.com/oracle/docker-images/blob/main/CONTRIBUTING.md#base-image-rules

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

File: OracleDatabase/SingleInstance/dockerfiles/21.3.0/setupLinuxEnv.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# Setup filesystem and oracle user
+# Adjust file permissions, go to /opt/oracle as user 'oracle' to proceed with Oracle installation
+# ------------------------------------------------------------
+mkdir -p $ORACLE_BASE/scripts/setup && \
+mkdir $ORACLE_BASE/scripts/startup && \
+ln -s $ORACLE_BASE/scripts /docker-entrypoint-initdb.d && \
+mkdir $ORACLE_BASE/oradata && \
+mkdir -p $ORACLE_HOME && \
+chmod ug+x $ORACLE_BASE/*.sh && \
+microdnf -y install oracle-database-preinstall-21c openssl && \
```
Comment (Djelibeybi) [inline OracleDatabase/SingleInstance/dockerfiles/21.3.0/setupLinuxEnv.sh:22]: After switching to `oraclelinux:8`, change this to `dnf` please.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

File: OracleDatabase/SingleInstance/dockerfiles/21.3.0/setupLinuxEnv.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+rm -rf /var/cache/yum && \
```
Comment (Djelibeybi) [inline OracleDatabase/SingleInstance/dockerfiles/21.3.0/setupLinuxEnv.sh:23]: This should be changed to `rm -rf /var/cache/dnf`

**Comment by gvenzl** [comment]
I believe this PR can be closed in favor of the already merged PR #2058. @abhisbyk, can you confirm once again, please?

**Comment by abhisbyk** [comment]
@gvenzl 
Yes, this PR can be closed

**Pull request closed without merging**